File: internal/tui/theme/theme.go
package theme

import "github.com/charmbracelet/lipgloss"

var (
	// Brand colors
	Blue      = lipgloss.Color("#5B9BD5")
	Cyan      = lipgloss.Color("#56CCF2")
	Green     = lipgloss.Color("#6BCB77")
	Red       = lipgloss.Color("#FF6B6B")
	Yellow    = lipgloss.Color("#FFD93D")
	Purple    = lipgloss.Color("#C084FC")
	Magenta   = lipgloss.Color("#F472B6")
	Muted     = lipgloss.Color("#555555")
	Dim       = lipgloss.Color("#3A3A3A")
	White     = lipgloss.Color("#FAFAFA")
	BgSubtle  = lipgloss.Color("#1A1A2E")
	BgPanel   = lipgloss.Color("#16213E")
	Highlight = lipgloss.Color("#0F3460")

	// Logo style
	Logo = lipgloss.NewStyle().
		Foreground(Purple).
		Bold(true)

	// Title styles
	Title = lipgloss.NewStyle().
		Bold(true).
		Foreground(Cyan)

	TitleAccent = lipgloss.NewStyle().
			Bold(true).
			Foreground(Purple)

	Subtitle = lipgloss.NewStyle().
			Foreground(Muted).
			Italic(true)

	// Status styles
	Success = lipgloss.NewStyle().
		Foreground(Green).
		Bold(true)

	Error = lipgloss.NewStyle().
		Foreground(Red).
		Bold(true)

	Warning = lipgloss.NewStyle().
		Foreground(Yellow)

	// Selection styles
	Selected = lipgloss.NewStyle().
			Foreground(Green).
			Bold(true)

	Unselected = lipgloss.NewStyle().
			Foreground(Muted)

	ActiveItem = lipgloss.NewStyle().
			Foreground(White).
			Bold(true)

	CursorStyle = lipgloss.NewStyle().
			Foreground(Cyan).
			Bold(true)

	// Spinner
	Spinner = lipgloss.NewStyle().
		Foreground(Purple)

	// Panel border
	Panel = lipgloss.NewStyle().
		Border(lipgloss.RoundedBorder()).
		BorderForeground(Highlight).
		Padding(1, 2)

	// Help bar
	HelpKey = lipgloss.NewStyle().
		Foreground(Purple).
		Bold(true)

	HelpSep = lipgloss.NewStyle().
		Foreground(Dim)

	HelpValue = lipgloss.NewStyle().
			Foreground(Muted)

	// Progress bar
	ProgressFilled = lipgloss.NewStyle().
			Foreground(Green)

	ProgressEmpty = lipgloss.NewStyle().
			Foreground(Dim)

	// Counter badge
	Badge = lipgloss.NewStyle().
		Foreground(Cyan).
		Bold(true)

	// Panel styles for runner side-by-side layout
	PanelBorder = lipgloss.NewStyle().
			Border(lipgloss.RoundedBorder()).
			BorderForeground(Highlight).
			Padding(0, 1)

	PanelTitle = lipgloss.NewStyle().
			Foreground(Muted).
			Italic(true)
)

// ProgressBar renders a progress bar with the given width and percentage.
func ProgressBar(width int, percent float64) string {
	filled := int(float64(width) * percent)
	if filled > width {
		filled = width
	}
	empty := width - filled
	bar := ProgressFilled.Render(repeatStr("█", filled)) +
		ProgressEmpty.Render(repeatStr("░", empty))
	return bar
}

func repeatStr(s string, n int) string {
	if n <= 0 {
		return ""
	}
	result := ""
	for i := 0; i < n; i++ {
		result += s
	}
	return result
}

File: crates/otobun/src/tui/theme.rs
use ratatui::style::{Color, Modifier, Style};

// Brand colors (matching Go theme)
pub const BLUE: Color = Color::Rgb(91, 155, 213);
pub const CYAN: Color = Color::Rgb(86, 204, 242);
pub const GREEN: Color = Color::Rgb(107, 203, 119);
pub const RED: Color = Color::Rgb(255, 107, 107);
pub const YELLOW: Color = Color::Rgb(255, 217, 61);
pub const PURPLE: Color = Color::Rgb(192, 132, 252);
pub const MAGENTA: Color = Color::Rgb(244, 114, 182);
pub const MUTED: Color = Color::Rgb(85, 85, 85);
pub const DIM: Color = Color::Rgb(58, 58, 58);
pub const WHITE: Color = Color::Rgb(250, 250, 250);
pub const BG_SUBTLE: Color = Color::Rgb(26, 26, 46);
pub const BG_PANEL: Color = Color::Rgb(22, 33, 62);
pub const HIGHLIGHT: Color = Color::Rgb(15, 52, 96);

pub fn logo() -> Style {
    Style::default().fg(PURPLE).add_modifier(Modifier::BOLD)
}

pub fn title() -> Style {
    Style::default().fg(CYAN).add_modifier(Modifier::BOLD)
}

pub fn title_accent() -> Style {
    Style::default().fg(PURPLE).add_modifier(Modifier::BOLD)
}

pub fn subtitle() -> Style {
    Style::default().fg(MUTED).add_modifier(Modifier::ITALIC)
}

pub fn success() -> Style {
    Style::default().fg(GREEN).add_modifier(Modifier::BOLD)
}

pub fn error() -> Style {
    Style::default().fg(RED).add_modifier(Modifier::BOLD)
}

pub fn warning() -> Style {
    Style::default().fg(YELLOW)
}

pub fn selected() -> Style {
    Style::default().fg(GREEN).add_modifier(Modifier::BOLD)
}

pub fn unselected() -> Style {
    Style::default().fg(MUTED)
}

pub fn active_item() -> Style {
    Style::default().fg(WHITE).add_modifier(Modifier::BOLD)
}

pub fn cursor() -> Style {
    Style::default().fg(CYAN).add_modifier(Modifier::BOLD)
}

pub fn spinner() -> Style {
    Style::default().fg(PURPLE)
}

pub fn help_key() -> Style {
    Style::default().fg(PURPLE).add_modifier(Modifier::BOLD)
}

pub fn help_sep() -> Style {
    Style::default().fg(DIM)
}

pub fn help_value() -> Style {
    Style::default().fg(MUTED)
}

pub fn badge() -> Style {
    Style::default().fg(CYAN).add_modifier(Modifier::BOLD)
}

pub fn progress_bar(width: u16, percent: f64) -> String {
    let filled = (width as f64 * percent).round() as usize;
    let empty = width as usize - filled;
    format!("{}{}", "█".repeat(filled), "░".repeat(empty))
}
